tests: i2c: i2c_target_api: added single mode testcase

CZKikin opened the pull request:
Some drivers do not support dual mode testcase. It would be great to test both signle and dual mode, even when the drivers do support this feature.

hakehuang commented on tests/drivers/i2c/i2c_target_api/testcase.yaml:
```diff
@@ -32,3 +32,10 @@ tests:
+     platform_allow:
+       - frdm_mcxn947/mcxn947/cpu0
+       - mimxrt1170_evk@B/mimxrt1176/cm7
+       - mimxrt1170_evk@B/mimxrt1176/cm4
```
how do you flash M4?

hakehuang commented:
I would not recommend to enable test this way,  as the board to board loopback testing is not that reliable, instead can you enable i2c_ram test through the ardiono_i2c?

dleach02 commented:
@CZKikin,  please address your CI issues

hakehuang commented on tests/drivers/i2c/i2c_target_api/testcase.yaml:
```diff
@@ -32,3 +32,10 @@ tests:
       - nucleo_f091rc
     extra_configs:
       - CONFIG_APP_DUAL_ROLE_I2C=y
+  drivers.i2c.target_api.single_role:
```
besides, look at the i2c_mcux_lpi2c.c if the CONFIG_I2C_TARGET is defined, it can support dual mode. why only enable single mode. @danieldegrasse can you confirm?

hakehuang commented on tests/drivers/i2c/i2c_target_api/testcase.yaml:
```diff
@@ -32,3 +32,10 @@ tests:
       - CONFIG_APP_DUAL_ROLE_I2C=y
+  drivers.i2c.target_api.single_role:
+     platform_allow:
+       - frdm_mcxn947/mcxn947/cpu0
```
for this board, you need also add overlay like what is existing in boards folder

hakehuang commented on tests/drivers/i2c/i2c_target_api/testcase.yaml:
```diff
@@ -32,3 +32,10 @@ tests:
+  drivers.i2c.target_api.single_role:
+     platform_allow:
+       - frdm_mcxn947/mcxn947/cpu0
+       - mimxrt1170_evk@B/mimxrt1176/cm7
```
I do not have revB but only RevA, which works fine in single mode. so please add mimxrt1170_evk/mimxrt1176/cm7, and its board overlay as well

hakehuang commented:
@CZKikin add mimxrt1170_evk/mimxrt1176/cm7 overlay, but do not remove the mimxrt1170_evk@B/mimxrt1176/cm7

Commit:
tests: i2c: i2c_target_api: added single mode testcase

Some drivers do not support dual mode testcase. It would be great to test
both signle and dual mode, even when the drivers do support this feature.

Signed-off-by: Krystof Sadlik <krystof.sadlik@nxp.com>

hakehuang commented:
board testing is PASS for all supporting platforms, but need fix the CI issue @CZKikin

danieldegrasse commented:
@CZKikin you can resolve the compliance error locally by amending your commit to use the same author name as the Signed-off entry, just run the following:
```
$ git commit --amend --author="Krystof Sadlik <krystof.sadlik@nxp.com>"     
```

Commit:
tests: i2c: i2c_target_api: fixed sizes in ovelay

Fixed sizes in ovelay files.

Signed-off-by: Krystof Sadlik <krystof.sadlik@nxp.com>

Commit:
tests: i2c: Fixed overlay for frdm board

Fixed overlay for frdm board as well.

Signed-off-by: Krystof Sadlik <krystof.sadlik@nxp.com>
